docs/qtpromise/qpromise/convert.md

```diff
@@ -0,0 +1,118 @@
+---
+title: .convert
+---
+
+# QPromise::convert
+
+*Since: 0.7.0*
+
+```cpp
+QPromise<T>::convert<U>() -> QPromise<U>
+```
+
+This method converts the resolved value of `QPromise<T>` to the type `U`. Depending on types `T`
+and `U`, it performs a [static cast](https://en.cppreference.com/w/cpp/language/static_cast), 
+calls a [converting constructor](https://en.cppreference.com/w/cpp/language/converting_constructor), 
+or tries to convert using [QVariant](https://doc.qt.io/qt-5/qvariant.html).
+
+If `T` and `U` are [fundamental types](https://en.cppreference.com/w/cpp/language/types) or 
+[enumerations](https://en.cppreference.com/w/cpp/language/enum), the result of the conversion is 
+the same as calling `static_cast<U>` for type `T`:
+
+```cpp
+QPromise<int> input = {...}
+
+// output type: QPromise<double>
+auto output = input.convert<double>();
+
+output.then([](double value) {
+    // the value has been converted using static_cast
+});
+```
+
+If `U` has a [converting constructor](https://en.cppreference.com/w/cpp/language/converting_constructor) 
```

Comment:
At first, I thought of `std::vector` accepting the size, but turns out this constructor is explicit: `https://en.cppreference.com/w/cpp/container/vector/vector`. 

Of course, this can happen if there is a constructor satisfying the converting constructor properties but not performing a conversion, and it probably might happen to some user classes (I think not everybody is strict with `explicit`). I believe for library classes we should trust the library developers to have produced the correct class definition, and for user classes we can't do anything except mentioning it in the docs. Anyway, I'm confident that QtPromise users are proficient enough to not expect some magical conversion to happen to their classes not having implemented it specifically in code :-)

I've elaborated in the docs on what a converting constructor is (because one extra click on the link is usually too many) and added a note in the docs regarding the correct behavior of the converting constructor.

Reply:
> because one extra click on the link is usually too many

Yes and it's also easier to understand the way you described it in the docs.

Thanks!